### PR TITLE
Add mqtt notify service and update example page

### PR DESCRIPTION
**Pull request by jbouwh**
## Proposed change

Add documentation for new MQTT notify platform.
Update the MQTT notify example page.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/64728
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/notify.mqtt.markdown [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+As alternative you can also choose to [publish notifications directly](/examples/notify.mqtt).
+
+{% configuration %}
+command_template:
+  description: Defines a [template](/docs/configuration/templating/#processing-incoming-data) to generate the payload to send to `command_topic`. The variable `value` will be assigned with the `message` parameter that was given when calling the notify service. The service parameters `message`, `name`, `service`, `title` and `target` can be used as variables in the template. Use the `data` service parameter to add additional variables.
+  required: false
+  type: template
+command_topic:
+  description: The MQTT topic to publish the notification message. To use `title`, `target` config or `data` service parameters in the payload, use `command_template` to format the payload.
+  required: false
```
Comment (emontnemery) [inline source/_integrations/notify.mqtt.markdown]: In code PR, this is required

File: source/_integrations/notify.mqtt.markdown [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  type: string
+device:
+  description: "Information about the device this device trigger is a part of to tie it into the [device registry](https://developers.home-assistant.io/docs/en/device_registry_index.html). At least one of identifiers or connections must be present to identify the device."
+  required: true
```
Comment (emontnemery) [inline source/_integrations/notify.mqtt.markdown]: In code PR, this is optional

File: source/_examples/notify.mqtt.markdown [hunk 3]
```diff
@@ -6,7 +6,15 @@ ha_category: Notifications
 ha_iot_class: Configurable
 ---
 
-The MQTT notification support is different than the other [notification](/integrations/notify/) platforms. It is a service. This means that you don't have to create a configuration entry but you need to provide more details when calling the service.
+The MQTT notification support is different than the other [notification](/integrations/notify/) platforms. It is a service.
+
+## Use the the MQTT notify platform
+
+The [MQTT notify service platform](/integrations/notify.mqtt/) allows you to use [MQTT discovery](/docs/mqtt/discovery/), or manual setup through `configuration.yaml` to enable a custom MQTT notify service. The notify platform helps to implement notification services on IoT notification screens, notification gateways, or for digital signage applications. An example is showing a text alert message on a climate control unit, e.g. an error, alert or informal message. The device can advertize the notification message service through discovery.
```
Comment (emontnemery) [inline source/_examples/notify.mqtt.markdown:13]: Based on the code PR, it seems like only setup through discovery is supported (there's no `async_setup_platform`)?

File: source/_integrations/notify.mqtt.markdown (hunk 1 above)
Comment (jbouwh) [reply source/_integrations/notify.mqtt.markdown]: updated in the docs

File: source/_integrations/notify.mqtt.markdown (hunk 2 above)
Comment (jbouwh) [reply source/_integrations/notify.mqtt.markdown]: updated

**Comment by jbouwh** [comment]
name is also required now

**Review by frenck — APPROVED** [review]
Thanks, @jbouwh 👍

**Pull request merged**